File: Cocoa/AppFrameworks/UIKit/TableView/TableCellObject/YJUITableCellObject.m
//
//  YJUITableCellObject.m
//  YJUITableViewFactory
//
//  HomePage:https://github.com/937447974/YJCocoa
//  YJ技术支持群:557445088
//
//  Created by 阳君 on 16/5/13.
//  Copyright © 2016年 YJCocoa. All rights reserved.
//

#import "YJUITableCellObject.h"
#import <objc/runtime.h>
#import "YJNSFoundationOther.h"

@implementation YJUITableCellObject

- (instancetype)initWithTableViewCellClass:(Class)cellClass {
    self = [super init];
    if (self) {
        _cellClass = cellClass;
        _cellName = YJNSStringFromClass(_cellClass);
        self.reuseIdentifier = _cellName;
    }
    return self;
}

@end

File: YJCocoa/Classes/AppFrameworks/UIKit/TableView/YJUITableCellObject.swift
//
//  YJUITableCellObject.swift
//  YJCocoa
//
//  HomePage:https://github.com/937447974/YJCocoa
//  YJ技术支持群:557445088
//
//  Created by 阳君 on 2019/5/22.
//  Copyright © 2016-现在 YJCocoa. All rights reserved.
//

import UIKit

/// 点击cell的回调
public typealias YJUITableCellSelectClosure = (_ tableViewManager: YJUITableViewManager, _ cellObject: YJUITableCellObject) -> Void

@objcMembers
open class YJUITableCellObject: NSObject {
    
    /// cell对应的VM
    public var cellModel: Any?
    /// 携带的自定义数据
    public var userInfo: Any?
    /// cell 点击回调
    public var didSelectClosure: YJUITableCellSelectClosure?
    
    /// cell 的类
    public private(set) var cellClass: AnyClass
    /// cell 的缓存，UITableViewCell.reuseIdentifier，默认类名
    public var reuseIdentifier = ""
    /// cell 的位置
    public var indexPath: IndexPath!
    /// cell 的高度
    public var height: CGFloat?
    
    public init(cellClass: AnyClass) {
        self.cellClass = cellClass
        self.reuseIdentifier = NSStringFromClass(cellClass)
    }
    
}
